content/kubermatic/master/tutorials_howtos/CCM_migration/CCM_migration_via_UI/_index.en.md

```diff
@@ -0,0 +1,60 @@
++++
+title = "CCM migration via UI"
+date = 2021-07-29T12:07:15+02:00
+weight = 10
+
++++
+
+This manual explains how to migrate to using external Cloud Controller Managers for supporting providers via UI.
+
+## Cloud Controller Manager (CCM)
+
+The [CCM](https://kubernetes.io/docs/concepts/architecture/cloud-controller/) (Cloud Controller Manager) is a Kubernetes
+control plane component that embeds cloud-specific control logic. There are two different kinds of Cloud controller managers:
+in-tree and out-of-tree. According to the Kubernetes [design proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers),
+the in-tree code is "code that lives in the core Kubernetes repository [k8s.io/kubernetes](https://github.com/kubernetes/kubernetes/)",
+while the out-of-tree code is "code that lives in an external repository outside of [k8s.io/kubernetes](https://github.com/kubernetes/kubernetes/)".
+
+The first cloud-specific interaction logic was completely in-tree, bringing in a not-negligible amount of problems,
+among which the dependency of the CCM release cycle from the Kubernetes core release cycle and the difficulty to add new providers
+to the Kubernetes core code. Then, the Kubernetes community moved toward the out-of-tree implementation by introducing
+a plugin mechanism that allows different cloud providers to integrate their platforms with Kubernetes.
+
+### CCM migration status
+To allow migration from in-tree to out-of-tree CCM for existing cluster, the cluster details view has been extended by a
+section in the top area, the "External CCM Migration Status", that indicates the status of the CCM migration.
+
+![Cluster Details View](/img/kubermatic/master/ui/ccm_migration_cluster_view.png?height=350px&classes=shadow,border "Cluster Details View")
```

Comment:
If you'd move `static/img/kubermatic/master/ui/ccm_migration_cluster_view.png` to this directory, it will be possible to do the following.

```suggestion
![Cluster Details View](ccm_migration_cluster_view.png?height=350px&classes=shadow,border "Cluster Details View")
```

Making it easy to manage in the future, when we will produce new copy (for 2.18 release) of master.

This is applicable to other part of those docs as well.